drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSourceLatestReleaseTestCase.java

```diff
@@ -0,0 +1,161 @@
+package org.jboss.arquillian.drone.webdriver.binary.downloading.source;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.CharStreams;
+import io.specto.hoverfly.junit.dsl.ResponseBuilder;
+import io.specto.hoverfly.junit.rule.HoverflyRule;
+import org.apache.http.HttpStatus;
+import org.assertj.core.api.JUnitSoftAssertions;
+import org.jboss.arquillian.drone.webdriver.binary.downloading.ExternalBinary;
+import org.jboss.arquillian.drone.webdriver.utils.GitHubLastUpdateCache;
+import org.jboss.arquillian.drone.webdriver.utils.HttpClient;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import javax.ws.rs.core.HttpHeaders;
+import java.io.*;
+
+import static io.specto.hoverfly.junit.core.SimulationSource.dsl;
+import static io.specto.hoverfly.junit.dsl.HoverflyDsl.service;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+public class GitHubSourceLatestReleaseTestCase extends HoverflyProxyHandler {
+
+    private static final String CACHED_CONTENT = "{\"lastModified\":\"Tue, 31 Jan 2017 17:16:07 GMT\"," +
+            "\"asset\":" +
+            "{\"version\":\"v0.14.0\"," +
+            "\"url\":\"https://github.com/mozilla/geckodriver/releases/download/v0.14.0/geckodriver-v0.14.0-linux64.tar.gz\"}" +
+            "}";
+
+    @ClassRule
+    public static HoverflyRule hoverflyRule = HoverflyRule.inSimulationMode();
+
+    @Rule
+    public final TemporaryFolder folder = new TemporaryFolder();
+
+    @Rule
+    public final JUnitSoftAssertions softly = new JUnitSoftAssertions();
+
+    private final String RESPONSE_BODY;
+    private File tmpFolder;
+    private GeckoDriverGitHubSource geckoDriverGitHubSource;
+    private HttpClient httpClientSpy;
+    private GitHubLastUpdateCache cacheSpy;
+
+    public GitHubSourceLatestReleaseTestCase() throws IOException {
+        this.RESPONSE_BODY = loadResponseBody("hoverfly/gh.simulation.mozilla@geckodriver.release.latest.json");
+    }
+
+    @Before
+    public void wireComponentsUnderTest() throws IOException {
+        this.tmpFolder = folder.newFolder();
+        this.httpClientSpy = spy(new HttpClient());
+        this.cacheSpy = spy(new GitHubLastUpdateCache(tmpFolder));
+        this.geckoDriverGitHubSource = new GeckoDriverGitHubSource(httpClientSpy, cacheSpy);
```

Comment:
Well it's not for the tests as such. It's just how objects are wired together. If we have constructor injection available here I'm happy to change it to this.